sentry-core/src/main/java/io/sentry/core/SentryEnvelopeHeaderAdapter.java

```diff
@@ -25,44 +28,64 @@ public void write(JsonWriter writer, SentryEnvelopeHeader value) throws IOExcept
       writer.value(value.getEventId().toString());
     }
 
-    final SdkInfo sdkInfo = value.getSdkInfo();
-    if (sdkInfo != null) {
-      boolean hasInitAttrs = false;
+    final SdkVersion sdkVersion = value.getSdkVersion();
+    if (sdkVersion != null) {
+      if (hasValidSdkVersion(sdkVersion)) {
+        writer.name("sdk").beginObject();
 
-      if (sdkInfo.getSdkName() != null) {
-        hasInitAttrs = initAttrs(writer, hasInitAttrs);
-        writer.name("sdk_name").value(sdkInfo.getSdkName());
+        writer.name("name").value(sdkVersion.getName());
+        writer.name("version").value(sdkVersion.getVersion());
 
-        // we only add versions if the name is not null
-        if (sdkInfo.getVersionMajor() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_major").value(sdkInfo.getVersionMajor());
-        }
-        if (sdkInfo.getVersionMinor() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_minor").value(sdkInfo.getVersionMinor());
-        }
-        if (sdkInfo.getVersionPatchlevel() != null) {
-          hasInitAttrs = initAttrs(writer, hasInitAttrs);
-          writer.name("version_patchlevel").value(sdkInfo.getVersionPatchlevel());
+        final List<String> integrations = sdkVersion.getIntegrations();
+        if (integrations != null) {
+          writer.name("integrations").beginArray();
+
+          for (final String item : integrations) {
+            writer.value(item);
```

Comment:
`l`: I prefer naming the items what they are. Makes the code easier to read IMO, but that's very nitpicking and personal preference. For `packages` below it makes more sense IMO, because when I look at `writer.name("name").value(item.getName());` I need to look 4 lines above to find out what the `item` is.
```suggestion
          for (final String integration : integrations) {
            writer.value(integration);
```

Reply:
I've renamed it but I am not a big fan of it because the list is the plural of the item and it can get confused to read

Reply:
As I said personal preference. Just wanted to point it out to learn from each other.